text/projects/Krzysztof_Michalik/Wee_LCP.tex

```diff
@@ -0,0 +1,179 @@
+\documentclass{article}
+\usepackage[utf8]{inputenc}
+
+\title{Wee LCP description}
+\author{krzysztof.michalik }
+\date{May 2022}
+
+\usepackage[T1]{fontenc}
+\usepackage[polish]{babel}
+\usepackage{amsfonts}
+\usepackage{amsmath}
+
+\usepackage{graphicx}
+\usepackage{tikz}
+\usetikzlibrary{positioning}
+\usepackage{geometry}
+
+\begin{document}
+
+\maketitle
+
+\section{Wstęp}
+
+Praca Johannesa Fishera zatytułowana Wee LCP przedstawia sposób reprezentowania struktury LCP w pamięci wykorzystującą $o(n) = O(\frac{n}{\log{\log{n}}})$ bitów, gdzie $n$ jest długością początkowego słowa $T$ dla którego obliczamy LCP.
+
+Autor zakłada dodatkowo, że:
+\begin{enumerate}
+    \item Dostępna jest tablica sufiksowa dla słowa $T$ oraz samo to słowo.
+    \item Dostęp do $i$-tej litery słowa $T$ zajmuje czas $O(1)$.
+    \item Dostęp do tablicy sufiksowej jest w czasie $t_A = \Omega(\log{n}^{\epsilon})$, gdzie $\epsilon$ jest pewną stałą.
+\end{enumerate}
+Ostatnie założenie bierze się stąd, że autor powołuje się na nie zwykłe tablice sufiksowe, które zajmują $n \lceil \log{n} \rceil$ bitów, lecz na ich skompresowaną wersję. [Nie jest podana dokładna liczba bitów zajmowana przez taką reprezentację.]
+
+Autor przedstawia dwa główne wyniki:
+\begin{enumerate}
+    \item Dostęp do LCP jest w czasie $O(t_A)$, podczas gdy sama struktura zajmuje $2n + O(\frac{n \log\log{n}}{\log{n}})$ bitów w pamięci.
+    \item Dostęp do LCP jest w czasie $O(t_A + \log(n)^{\delta})$, podczas gdy sama struktura zajmuje $o(n) = O(\frac{n}{\log{\log{n}}})$ bitów w pamięci.
+\end{enumerate}
+
+W obu przypadkach osobno dostępny jest tekst $T$ oraz tablica sufiksowa $SUF$, a ograniczenia na pamięć dotyczą wyłącznie samego sposobu reprezentowania tablicy LCP.
+
+Warto zwrócić uwagę, że mimo zwiększonego czasu odpowiedzi na zapytanie w drugim z przedstawionych wyników, jest on w pewnym sensie podobny do czasu $t_A$ - oba są postaci $\log{n}^x$, gdzie w jedym przypadku $x = \epsilon$, a w drugim $x = \delta$. Oznacza to, że odpowiedni dobór $\delta$ asymptotycznie doprowadzi do tego samego czasu odpowiedzi na zapytanie do LCP.
+
+\section{Oznaczenia}
+
+Przez $T$ będziemy oznaczać tekst, który dostajemy na wejściu.
+
+Przez $T[i:]$ będziemy oznaczać sufiks słowa $T$ zaczynający się na pozycji $i$.
+
+Jeśli nie będziemy znali indeksu na którym zaczyna się sufiks, który jest $a$-ty leksykograficznie, to przez $S_a$ będziemy oznaczać ten sufiks.
+
+$S_A$ będzie tablicą sufiksową, a dokładniej $S_A[i] = j$ będzie oznaczało, że $T[i:]$ jest $j$-tym leksykograficznie sufiksem słowa $T$.
```

Comment:
Sometimes you denote suffix array with $S_A$, sometimes with $SUF$ - please unify